src/movies.js

```diff
@@ -1,25 +1,192 @@
 // Iteration 1: All directors? - Get the array of all directors.
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+/*
+{
+  title: 'The Shawshank Redemption',
+  year: 1994,
+  director: 'Frank Darabont',
+  duration: '2h 22min',
+  genre: ['Crime', 'Drama'],
+  score: 9.3
+},
+*/
+function getAllDirectors(moviesArray) {
+  return moviesArray.map(el => el.director);
+
+  /*****  returning the array filtered. ******
+  return moviesArray
+    .map(el => el.director)
+    .filter((director, index, self) => self.indexOf(director) === index);
+  */
+}
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+function howManyMovies(moviesArray) {
+  return moviesArray.filter(el => el.director === 'Steven Spielberg' && el.genre.includes('Drama')).length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+function scoresAverage(moviesArray) {
+  let len = moviesArray.length;
+
+  if (len === 0) {
+    return 0;
+  }
+
+  const number = moviesArray.reduce((acc, el) => {
+    if (!el.score) {
+      return acc + 0;
+    } else {
+      return acc + el.score;
+    }
+  }, 0);
+
+  return +(number/len).toFixed(2);
+}
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+function dramaMoviesScore(moviesArray) {
+  const dramaMovies = moviesArray.filter(el => el.genre.includes('Drama'));
+
+  if (dramaMovies.length === 0) {
+    return 0;
+  }
+
+  const score = dramaMovies.reduce((acc, elem) => {
+    return acc + elem.score;
+  }, 0);
+
+  return parseFloat((score / dramaMovies.length).toFixed(2));
```

Comment:
Esto se parece muy a scoreAverage no? yo lo haria asi entonces!

```suggestion
  return scoreAverage(dramaMovies);
```